controllers/awsmachine_controller.go

```diff
@@ -411,7 +411,12 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	}
 
 	// Make sure Spec.ProviderID is always set.
-	machineScope.SetProviderID(fmt.Sprintf("aws:////%s", instance.ID))
+	failureDomain := machineScope.AWSMachine.Spec.FailureDomain
```

Comment:
The particular provider failureDomain field is to be deprecated in favour of the core machine FailureDomain. So that one needs to be favoured as in
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/425e1bb337cfb76ceb38c1cce97345c26fafe08d/pkg/cloud/services/ec2/instances.go#L162-L165